src/chart/generateCategoricalChart.tsx

```diff
@@ -2247,9 +2247,10 @@ export const generateCategoricalChart = ({
               return { graphicalItem, payload: activeBarItem };
             }
           } else if (
-            isFunnel(graphicalItem, activeItem) ||
-            isPie(graphicalItem, activeItem) ||
-            isScatter(graphicalItem, activeItem)
+            activeItem != null &&
+            (isFunnel(graphicalItem, activeItem) ||
+              isPie(graphicalItem, activeItem) ||
+              isScatter(graphicalItem, activeItem))
```

Comment:
instead of doing it this way could we add the check in each `is..` function?

That way this is never an issue elsewhere that those typeguards are used.

so for example
```ts
 export function isScatter(graphicalItem: GraphicalItem, _item: unknown): _item is ScatterItem { 
   return Boolean(_item) && 'points' in graphicalItem.props; 
 } 
```